.github/workflows/deploy.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      url:
+        required: true
+        type: string
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    environment:
+      name: ${{ inputs.environment }}
+      url: ${{ inputs.url }}
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Restore Build Cache
+      id: restore-build-cache
+      uses: actions/cache/restore@v4
+      with:
+        path: |
+          ~/.cache/Cypress
+          common/temp
+          */.rush/temp
+          */node_modules
+          */dist
+        key: ${{ runner.os }}-build-${{ hashFiles('common/config/rush/repo-state.json') }}
+        fail-on-cache-miss: true
+
+    - name: Configure Discord Secrets
+      run: (cd discord && ./client_config_ci.sh)
+      env:
+        # Repository-level secrets
+        DISCORD_CLIENT_ID: ${{ secrets.DISCORD_CLIENT_ID }}
+        DISCORD_CLIENT_SECRET: ${{ secrets.DISCORD_CLIENT_SECRET }}
+        DISCORD_SESSION_SECRET: ${{ secrets.DISCORD_SESSION_SECRET }}
+        # Secrets overridden by the workflow environment
+        DISCORD_GUILD_ID: ${{ secrets.DISCORD_GUILD_ID }}
+        DISCORD_REDIRECT_URI: ${{ secrets.DISCORD_REDIRECT_URI }}
+
+    - name: Configure Deploy Secrets
+      run: (cd deploy && ./client_config_ci.sh)
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        ENVIRONMENT: ${{ inputs.environment }}
+
+    - name: Create Deployment Artifacts
+      run: node common/scripts/install-run-rush.js deploy --create-archive deploy.zip
+
+    - name: Upload Deployment Artifacts
+      run: (cd deploy && npm run upload ${{ inputs.environment }})
+      # TODO: any utility in uploading deploy.zip with actions/upload-artifact? https://github.com/actions/upload-artifact
```

Comment:
already uploading build artifacts to aws s3, would it make sense to replace (or duplicate??) that with actions/upload-artifact?

probably not? s3 download via ec2 instance user data script already works